File: ABC107_C.cpp
# include<bits/stdc++.h>
using namespace std;

int INF = 1001001001;

int main() { 
    int N, K;
    cin >> N >> K;
    
    vector<int> x(N);
    for (int i = 0; i < N; i++) {
        cin >> x[i];
    }

    int left = 0, right = K - 1, time = INF;
    while(right < N) {
        time = min(time, abs(x[left]) +abs(x[right] - x[left]));
        time = min(time, abs(x[right]) + abs(x[left] - x[right]));
        left ++;
        right ++;
    }

    cout << time << endl;

}

File: ABC107_C.py
import sys


def movingDistance(left, right):
    if right <= 0:
        return abs(left)
    if left >= 0:
        return right

    return min(abs(left) * 2 + right, abs(left) + right * 2)


def main():
    N, K = map(int, sys.stdin.readline().strip().split())
    X = list(map(int, sys.stdin.readline().strip().split()))

    if N == K:
        print(movingDistance(left=X[0], right=X[-1]))
        return

    r = K - 1
    ans = 1001001001
    while r < N:
        ans = min(ans, movingDistance(left=X[r - K + 1], right=X[r]))
        r += 1

    print(ans)
    return


if __name__ == "__main__":
    main()
